Makefile

```diff
@@ -38,17 +56,80 @@ checkvars:
 
 setup: pilot/platform/kube/config
 
-check:
-	echo 'To be added'
+#-----------------------------------------------------------------------------
+# Target: depend
+#-----------------------------------------------------------------------------
+.PHONY: depend 
+.PHONY: depend.status depend.ensure depend.graph
+
+depend: depend.ensure
+
+Gopkg.lock: Gopkg.toml ; $(info $(H) generating) @
+	$(Q) dep ensure -update
+
+depend.status: Gopkg.lock ; $(info $(H) reporting dependencies status...)
+	$(Q) dep status
+
+# @todo only run if there are changes (e.g., create a checksum file?) 
+depend.ensure: Gopkg.lock ; $(info $(H) ensuring dependencies are up to date...)
+	$(Q) dep ensure
+
+depend.graph: Gopkg.lock ; $(info $(H) visualizing dependency graph...)
+	$(Q) dep status -dot | dot -T png | display
+
+#-----------------------------------------------------------------------------
+# Target: precommit
+#-----------------------------------------------------------------------------
+.PHONY: precommit format check
+.PHONY: fmt format.gofmt format.goimports format.bazel
+.PHONY: check.vet check.lint
+
+precommit: format check
+format: format.goimports
+fmt: format.gofmt format.goimports format.bazel # backward compatible with ./bin/fmt.sh
+check: check.vet check.lint
+
+format.gofmt: ; $(info $(H) formatting files with go fmt...)
+	$(Q) gofmt -s -w $(GO_FILES)
+
+format.goimports: ; $(info $(H) formatting files with goimports...)
+	$(Q) goimports -w -local istio.io $(GO_FILES)
+
+format.bazel: ; $(info $(H) formatting bazel files...)
+	$(eval BAZEL_FILES = $(shell git ls-files | grep -e 'BUILD' -e 'WORKSPACE' -e 'BUILD.bazel' -e '.*\.bazel' -e '.*\.bzl'))
+	$(Q) buildifier -mode=fix $(BAZEL_FILES)
+
+# @todo fail on vet errors? Currently uses `true` to avoid aborting on failure
+check.vet: ; $(info $(H) running go vet on packages...)
+	$(Q) $(GO) vet $(PACKAGES) || true
+
+# @todo fail on lint errors? Currently uses `true` to avoid aborting on failure
+# @todo remove _test and mock_ from ignore list and fix the errors?
+check.lint: ; $(info $(H) running golint on packages...)
+	$(eval LINT_EXCLUDE := $(GO_EXCLUDE)|_test.go|mock_)
+	$(Q) for p in $(PACKAGES); do \
+		golint $$p | grep -v -E '$(LINT_EXCLUDE)' ; \
+	done || true;
+
+# @todo gometalinter targets?
 
 build: setup
 	bazel $(BAZEL_STARTUP_ARGS) build $(BAZEL_BUILD_ARGS) //...
 
-fmt:
-	bin/fmt.sh
+#-----------------------------------------------------------------------------
+# Target: precommit
+#-----------------------------------------------------------------------------
+.PHONY: clean
+.PHONY: clean.bazel clean.go
+
+clean: clean.bazel
```

Comment:
shouldn't this be clean.go?

Reply:
Keeping the clean target backward compatible. Can always run `make clean.go` directly